plugins/kafka-backend/src/service/KafkaApi.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Kafka, SeekEntry } from 'kafkajs';
+import { Logger } from 'winston';
+
+export type PartitionOffset = {
+  id: number;
+  offset: string;
+};
+
+export type TopicOffset = {
+  topic: string;
+  partitions: PartitionOffset[];
+};
+
+export class KafkaApi {
+  private readonly kafka: Kafka;
+  private readonly logger: Logger;
+
+  constructor(clientId: string, brokers: string[], logger: Logger) {
+    logger.info(
+      `creating kafka client with clientId=${clientId} and brokers=${brokers}`,
+    );
+
+    this.kafka = new Kafka({ clientId, brokers });
+    this.logger = logger;
+  }
+
+  async fetchTopicOffsets(topic: string): Promise<Array<PartitionOffset>> {
+    this.logger.info(`fetching topic offsets for ${topic}`);
```

Comment:
maybe look through all the loggings and make them just `debug` - no need to have this in the normal log flow right?